CMakeLists.txt

```diff
@@ -97,23 +165,31 @@ set(CHPL_QTHREAD_UNIQ_CFG_PATH CACHE INTERNAL "Description")
 set(CHPL_RE2_UNIQ_CFG_PATH CACHE INTERNAL "Description")
 set(CHPL_PE_CHPL_PKGCONFIG_LIBS CACHE INTERNAL "Description")
 
+
 # Set ENV versions of CHPL_ vars that are not in printchplenv output, in the
 # case when they are set from the cmdline, otherwise get them from the environment
 if (NOT DEFINED ${CHPL_DEVELOPER})
-  message(NOTICE "CHPL_DEVELOPER not given, using $ENV{CHPL_DEVELOPER}")
-  set(CHPL_DEVELOPER $ENV{CHPL_DEVELOPER} CACHE BOOL "Override CHPL_DEVELOPER flag" FORCE)
+  if(DEFINED ENV{CHPL_DEVELOPER})
+    set(CHPL_DEVELOPER $ENV{CHPL_DEVELOPER} CACHE BOOL "Override CHPL_DEVELOPER flag" FORCE)
+    message(NOTICE "CHPL_DEVELOPER found in ENV, using $ENV{CHPL_DEVELOPER}")
+  else()
+    set(CHPL_DEVELOPER OFF CACHE BOOL "Override CHPL_DEVELOPER flag" FORCE)
+    message(NOTICE "CHPL_DEVELOPER is not set, using ${CHPL_DEVELOPER}")
+  endif()
 else()
   set(ENV{CHPL_DEVELOPER} ${CHPL_DEVELOPER})
```

Comment:
I believe it's this way so that when we call to printchplenv later, the return values it gives us don't conflict with any cmake settings we passed in

Reply:
If they have to all line up, it seems like it would just be equivalent for us to use the environment variable only and not have the CMake `CHPL_DEVELOPER` variable.